File: DSA/done/3658 - gcdOfOddEvenSums.cpp
// https://leetcode.com/problems/gcd-of-odd-and-even-sums/

class Solution {
public:
    int gcd(int x, int y) { return y == 0 ? x : gcd(y, x % y); }
    int gcdOfOddEvenSums(int n) { return gcd(n * n, n * (n + 1)); }
};

File: DSA/done/3658 - gcdOfOddEvenSums.py
# https://leetcode.com/problems/gcd-of-odd-and-even-sums/

class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        def gcd(x: int, y: int) -> int:
            return x if y == 0 else gcd(y, x % y)

        return gcd(n * n, n * (n + 1))